[WIP] Allow to show a table in the IPython notebook using dataTables.

saimn opened the pull request:
This feature existed but was broken (https://github.com/astropy/astropy/issues/3937#issuecomment-131956530). cc @taldcroft @keflavich 
Here a new method `show_in_notebook` is added (it could replace the `_repr_html` but it is probably safer to use a separate method for now). 
The dataTables js file is loaded with requirejs, and a lightweight css style is added using Bootstrap. About the CSS:
- we could use directly the notebook's one but it is not very pretty,
- here I use Bootstrap which is already available in the notebook, ~~so I need to add classes to the HTML table which I do with a ugly `.replace`, it would be cleaner with #4131~~ (done).
- it could also be possible to use the dataTables CSS, but it requires one more external download and it seems not necessary.

![ipynb](https://cloud.githubusercontent.com/assets/311639/10149989/6717d53a-663d-11e5-9e8c-ac6e2d70ff31.png)

keflavich commented:
Looks neat.  Are there any weird interactions with ipython's mapping of keyboard commands/shortcuts?  I vaguely remember having some issues with that.

saimn commented:
Rebased and updated to use #4131

@keflavich It doesn't seem that dataTables binds some keyboard event by default, and I didn't find something about this.

astrofrog commented:
@saimn - can you rebase and add a changelog entry in the 1.2 section? (sorry that this didn't get merged in time for 1.1!)

saimn commented:
@astrofrog done.

eteq commented:
@saimn - because this is actually a fix in that it reactivated a feature that previously existed, we decided it was ok to treat this as a bug fix and merge it for 1.1 (I updated the changelog in the manual merge).

Thanks!

saimn commented:
Great :-), thanks @eteq !

eteq commented:
backported to v1.1.x in 9889c7722a24ce84870f7e6a21d005190f4c5163
